os/include/net/lwip/lwipopts.h

```diff
@@ -81,6 +81,12 @@
 #define LWIP_DHCP_TCPIP_THREAD 0
 #endif
 
+#ifdef CONFIG_LWIP_DHCP_HOSTNAME
+#define LWIP_NETIF_HOSTNAME 1
```

Comment:
Why can't you use CONFIG_LWIP_DHCP_HOSTNAME directly instead of this definition? If LWIP_NETIF_HOSTNAME's type is integer, then it is reasonable but it is kind of flag, In my opinion.

Reply:
Based on the configuration management in LwIP, LWIP_NETIF_HOSTNAME should be defined with a value of 1 or 0. However, CONFIG_LWIP_DHCP_HOSTNAME is defined in os/net/Kconfig, which is bool variable. Therefore, if CONFIG_LWIP_DHCP_HOSTNAME is not defined, LWIP_NETIF_HOSTNAME should be set to 0. If it is defined, it does not matter to assign 1 or CONFIG_LWIP_DHCP_HOSTNAME to LWIP_NETIF_HOSTNAME. Thanks for the comment.

Anyway, considering the possibility of CONFIG_LWIP_DHCP_HOSTNAME is set to 0, I applied your comment. Thanks again!